client.go

```diff
@@ -578,3 +627,53 @@ type StatusMeta struct {
 	InverseEnabled bool
 	TimeQuantum    string
 }
+
+type viewsInfo struct {
+	Views []string `json:"views"`
+}
+
+type exportReader struct {
+	sliceURIs    map[uint64]*URI
+	frame        *Frame
+	body         []byte
+	bodyIndex    int
+	currentSlice uint64
+	sliceCount   uint64
+	view         string
+}
+
+func newExportReader(sliceURIs map[uint64]*URI, frame *Frame, view string) *exportReader {
+	return &exportReader{
+		sliceURIs:  sliceURIs,
+		frame:      frame,
+		sliceCount: uint64(len(sliceURIs)),
+		view:       view,
+	}
+}
+
+func (r *exportReader) Read(p []byte) (n int, err error) {
+	if r.currentSlice >= r.sliceCount {
+		return 0, io.EOF
+	}
+	if r.body == nil {
+		uri, _ := r.sliceURIs[r.currentSlice]
+		headers := map[string]string{
+			"Accept": "text/csv",
+		}
+		client := NewClientWithURI(uri)
+		path := fmt.Sprintf("/export?index=%s&frame=%s&slice=%d&view=%s",
+			r.frame.index.Name(), r.frame.Name(), r.currentSlice, r.view)
+		_, r.body, err = client.httpRequest("GET", path, nil, headers, errorCheckedResponse)
+		if err != nil {
+			return 0, err
+		}
+		r.bodyIndex = 0
+	}
+	n = copy(p, r.body[r.bodyIndex:])
+	r.bodyIndex += n
+	if n >= len(r.body) {
+		r.body = nil
+		r.currentSlice++
+	}
+	return
```

Comment:
The `Read` function's signature is direct from Go docs: https://golang.org/pkg/io/#Reader so I thought its value would be obvious. Removed unnamed return values.